webapp/src/app/bcviewer/bcviewer.component.ts

```diff
@@ -188,13 +188,27 @@ class LinkBlock {
     index: number;
     height: number;
     maxHeight: number;
+    sign: string;
     id: Buffer;
 
-    constructor(sbRPC: SkipchainRPC, link: (ForwardLink | Buffer)) {
+    constructor(sbRPC: SkipchainRPC, link: (ForwardLink | Buffer), sbNow?: SkipBlock) {
         if (Buffer.isBuffer(link)) {
             this.id = link as Buffer;
         } else {
-            this.id = (link as ForwardLink).to;
+            const l = (link as ForwardLink);
+            this.id = l.to;
+            if (sbNow !== undefined) {
+                this.sign = "";
+                // TODO: extend to more than 32 nodes
+                const maskBuf = Buffer.alloc(4);
+                l.signature.getMask().copy(maskBuf);
+                const mask = Buffer.from(maskBuf.reverse()).readInt32BE(0);
+                const roster = l.newRoster || sbNow.roster;
```

Comment:
In this case, it would be better to use the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_Coalescing_Operator) but that's not yet stable. It has the same kind of issue than using plain `if` to check for `undefined`: the left operand is casted to a boolean and equated to `false`, which would be the case for `undefined` and `null`, but also for `NaN`, `0`, `""`, ...

So, for now, better explicilty check for `undefined`
```suggestion
                const roster = l.newRoster !== undefined ? l.newRoster : sbNow.roster;
```

Reply:
hmm - as `newRoster` is either undefined or of type `Roster`, I don't see the problem here. In the definition in `skipblock.ts` it should be `Roster | undefined`.

Reply:
> In the definition in `skipblock.ts` it should be `Roster | undefined`.

I only trust explicit types :P but yeah, that won't realistically happen.